src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1435,19 +1435,22 @@ private CSharpSyntaxNode GetBindingRoot(CSharpSyntaxNode node)
             }
 #endif
 
-            for (CSharpSyntaxNode current = node; current != this.Root; current = current.ParentOrStructuredTriviaParent)
+            if (EnableNullableAnalysis)
             {
-                if (current is StatementSyntax)
-                {
-                    return current;
-                }
+                // PROTOTYPE(nullable-api): This doesn't work for speculative models: the root is just the statement
+                // or expression being speculated on.
+                return Root;
+            }
 
-                switch (current.Kind())
+            for (CSharpSyntaxNode current = node; current != this.Root; current = current.ParentOrStructuredTriviaParent)
+            {
+                switch (current)
```

Comment:
>switch (current) [](start = 16, length = 16)

It is not clear what has motivated the refactoring inside this loop. It doesn't look like there is an intent to change a behavior. Consider reverting.

Reply:
There was initially an intent to change behavior, and this refactoring made that much clearer. After that was ruled out, I found that this version was clearer than the existing code, so I kept it.

---
In reply to: [269801499](https://github.com/dotnet/roslyn/pull/34429#discussion_r269801499) [](ancestors = 269801499)